packages/contracts-bedrock/test/L1/OPPrestateUpdater.t.sol

```diff
@@ -0,0 +1,352 @@
+                proxyAdmin: IProxyAdmin(address(0)),
+                absolutePrestate: Claim.wrap(0)
+            })
+        );
+
+        vm.expectRevert(OPPrestateUpdater.NotImplemented.selector);
+        prestateUpdater.upgrade(opChainConfigs);
+    }
+
+    function test_addGameType_notImplemented_reverts() public {
+        gameInput.push(
+            OPContractsManager.AddGameInput({
+                saltMixer: "hello",
+                systemConfig: ISystemConfig(address(0)),
+                proxyAdmin: IProxyAdmin(address(0)),
+                delayedWETH: IDelayedWETH(payable(address(0))),
+                disputeGameType: GameType.wrap(2000),
+                disputeAbsolutePrestate: Claim.wrap(bytes32(hex"deadbeef1234")),
+                disputeMaxGameDepth: 73,
+                disputeSplitDepth: 30,
+                disputeClockExtension: Duration.wrap(10800),
+                disputeMaxClockDuration: Duration.wrap(302400),
+                initialBond: 1 ether,
+                vm: IBigStepper(address(opcm.implementations().mipsImpl)),
+                permissioned: true
+            })
+        );
+
+        vm.expectRevert(OPPrestateUpdater.NotImplemented.selector);
+        prestateUpdater.addGameType(gameInput);
+    }
+
+    function test_l1ContractsRelease_works() public view {
+        string memory result = "none";
+
+        assertEq(result, prestateUpdater.l1ContractsRelease());
+    }
+
+    function addGameType(IOPContractsManager.AddGameInput memory input)
+        internal
+        returns (IOPContractsManager.AddGameOutput memory)
+    {
+        IOPContractsManager.AddGameInput[] memory inputs = new IOPContractsManager.AddGameInput[](1);
+        inputs[0] = input;
+
+        (bool success, bytes memory rawGameOut) =
+            address(opcm).delegatecall(abi.encodeCall(IOPContractsManager.addGameType, (inputs)));
+        assertTrue(success, "addGameType failed");
+
+        IOPContractsManager.AddGameOutput[] memory addGameOutAll =
+            abi.decode(rawGameOut, (IOPContractsManager.AddGameOutput[]));
+        return addGameOutAll[0];
+    }
+
+    function newGameInputFactory(bool permissioned) internal view returns (IOPContractsManager.AddGameInput memory) {
+        return IOPContractsManager.AddGameInput({
+            saltMixer: "hello",
+            systemConfig: chainDeployOutput.systemConfigProxy,
+            proxyAdmin: chainDeployOutput.opChainProxyAdmin,
+            delayedWETH: IDelayedWETH(payable(address(0))),
+            disputeGameType: GameType.wrap(2000),
+            disputeAbsolutePrestate: Claim.wrap(bytes32(hex"deadbeef1234")),
+            disputeMaxGameDepth: 73,
+            disputeSplitDepth: 30,
+            disputeClockExtension: Duration.wrap(10800),
+            disputeMaxClockDuration: Duration.wrap(302400),
+            initialBond: 1 ether,
+            vm: IBigStepper(address(opcm.implementations().mipsImpl)),
```

Comment:
```suggestion
            vm: IBigStepper(address(0)),
```

Reply:
If I sent address 0 it reverts with DeploymentFailed

Reply:
If I send address 0 it reverts with DeploymentFailed

Reply:
resolving, can discuss above.